Design/histogram_type.md

```diff
@@ -0,0 +1,108 @@
+# Histogram Data Type
+
+**Note:** This document is mainly concerned with a data type for a *generic* histogram.
+In the context considered here, a histogram is *not* specific to neutron science and probably not even specific to physics.
+In particular it should not be confused with the current data types in Mantid that are joint containers for histogram data and spectrum/detector information, such as `ISpectrum` and `Histogram1D`.
+
+
+## Motivation
+
+One of the most fundamental concepts of ubiquitous use in Mantid is that of histograms of (event) data.
+Despite this, there is currently no data type in Mantid that deals with histogram data.
+The purpose of this design proposal is to fix this.
+
+There are various motivations for this:
+
+- Reduce code duplication by providing arithmetic operations such as addition of histograms.
+- Hide handling of errors (uncertainties).
+- Hide discrimination between histogram and distribution data.
+- Decrease risk for certain types of bugs by using the C++ type system.
+
+All this would remove duplicated code from the client and put it into a central place where it can be unit-tested.
+
+
+## Current situation
+
+The current way of dealing with histograms has various shortcomings.
+Basically we have a set of three `std::vector<double>`, for `X`, `Y`, and `E`.
+There is a single `typedef` `MantidVec` used for all three of them (and other data), and this is probably adding only more confusion.
+
+The problems include:
+
+- No type safety:
```

Comment:
I think what you linked to goes one step further than I intended at this point. My suggestion was to have, e.g., a type for the *vector* of `Y` values, *not* for the *individual* `Y` values of each bin. The latter would certainly be a good idea if we continue to have plenty of histogram-modifying code spread in the code base. However, the goal of this design is to move such code to a central point, so at this point is it not clear if the benefits of having a type for each individual value outweigh the increased complexity.